testlapack/general.go

```diff
@@ -594,3 +594,55 @@ func isOrthonormal(q blas64.General) bool {
 	}
 	return true
 }
+
+// copyMatrix copies an m×n matrix src of stride n into an m×n matrix dst of stride ld.
+func copyMatrix(m, n int, dst []float64, ld int, src []float64) {
+	for i := 0; i < m; i++ {
+		copy(dst[i*ld:i*ld+n], src[i*n:i*n+n])
+	}
+}
+
+// equalApprox returns whether the matrices A and B of order n are approximately
```

Comment:
Systematic rather than random - I used to do random, but I was convinced by your arguments, and have seen the benefits in my own debugging sessions.